Tests/Unit/MSFT_xWebAppPool.Tests.ps1

```diff
@@ -0,0 +1,862 @@
+
+            .PARAMETER appcmdVal
+            The location that would be used to get this value via appcmd.
+            
+            .PARAMETER value1
+            A valid value that $option can be set to.  Must be different than value2
+
+            .PARAMETER value2
+            A valid value that $option can be set to.  Must be different than value1
+
+            .EXAMPLE
+            SetTargetResource "autoStart" "add.autoStart" "true" "false"
+
+            #>
+            [OutputType([bool])]
+            Param($option,$appcmdVal,$value1,$value2)
+            $testParams =@{
+                Name = "DefaultAppPool"
+                Ensure = "Present"
+            }
+            $curVal = Invoke-Expression "(([xml](& $env:SystemRoot\system32\inetsrv\appcmd.exe list apppool 'DefaultAppPool' /config:*)).$($appcmdVal))"
+            if($curVal -ne $value1)
+            {
+                $testParams.Add($option,$value1)
+                Set-TargetResource @testParams | Out-Null #So we only return the value of Test-TargetResource below
+                $curVal = Invoke-Expression "(([xml](& $env:SystemRoot\system32\inetsrv\appcmd.exe list apppool 'DefaultAppPool' /config:*)).$($appcmdVal))"
+                $value1 -eq $curVal
+            }
+            else
+            {
+                $testParams.Add($option,$value2)
+                Set-TargetResource @testParams | Out-Null #So we only return the value of Test-TargetResource below
+                $curVal = Invoke-Expression "(([xml](& $env:SystemRoot\system32\inetsrv\appcmd.exe list apppool 'DefaultAppPool' /config:*)).$($appcmdVal))"
+                $value2 -eq $curVal
+            }
+
+            Return 
+        }
+
+        try
+        {
+            $baseParams =@{
+                Name = "DefaultAppPool"
+                Ensure = "Present"
+            }
+
+            It 'DefaultAppPool should exist' {           
+                (Get-ChildItem IIS:\apppools).Name.Contains("DefaultAppPool") | Should be $true
+            }
+
+            Context 'Test-TargetResource' {
+                
+                It 'Passes autoStart Test when same' {
+                    
+                    TestTargetResourceSame "autoStart" "add.autoStart" | Should be $true
+                }
+
+                It 'Fails autoStart Test when different' {
+                    
+                    TestTargetResourceDiff "autoStart" "add.autoStart" "true" "false"| Should be $false
+                }
+
+                It 'Passes Runtime Version Test when same' {
+                  
+                    TestTargetResourceSame "managedRuntimeVersion" "add.managedRuntimeVersion"  | Should be $true
+                }
+
+                It 'Fails Runtime Version Test when different' {
+                  
+                    TestTargetResourceDiff "managedRuntimeVersion" "add.managedRuntimeVersion" "v4.0" "v2.0" | Should be $false
```

Comment:
This becomes
Test-TargetResource -ManagedRunTimeVersion '2.0' | Should Be $False

This passes because $global:PoolCfg is set to v4.0.